src/components/chip/selectable-chip/index.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  StandaloneChevronDownRegularIcon,
+  StandaloneCircleXmarkRegularIcon,
+} from '@deriv/quill-icons'
+import { forwardRef } from 'react'
+import {
+  ChipIconSizes,
+  chipBaseClassnames,
+  ChipStandaloneIconSizes,
+  chipBaseVariant,
+  chipSelectionClassnames,
+} from '../chip.classnames'
+import { SelectableChipProps } from '../types'
+import qtMerge from 'qtMerge'
+
+export const SelectableChip = forwardRef<
+  HTMLButtonElement,
+  SelectableChipProps
+>(
+  (
+    {
+      className,
+      children,
+      dismissible = false,
+      icon: Icon,
+      isDropdownOpen,
+      dropdown = false,
+      labelTag,
+      onChipSelect,
+      onDismiss,
+      size,
+      ...rest
+    },
+    ref,
+  ) => {
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+      if (dismissible || dropdown) return
+      const target = event.currentTarget
+      const isSelected = target.getAttribute('data-state') === 'selected'
+      const selected_state = isSelected ? '' : 'selected'
+      target.setAttribute('data-state', selected_state)
+      onChipSelect?.(event, !isSelected)
+    }
+
+    const handleDismiss = (
+      event: React.MouseEvent<SVGSVGElement, MouseEvent>,
+    ) => {
+      onDismiss?.(event)
+    }
+
+    return (
+      <button
+        className={chipBaseVariant({
+          dismissible,
+          size,
+          className: qtMerge(
+            chipBaseClassnames,
+            chipSelectionClassnames,
+            className,
+          ),
+        })}
+        data-state=""
+        ref={ref}
+        onClick={handleClick}
+        {...rest}
+      >
+        {Icon && <Icon {...ChipIconSizes[size ?? 'md']} />}
```

Comment:
```suggestion
        {Icon && <Icon iconSize={size} />}
```

Reply:
@jim-deriv currently the iconSize is not accurate. I have discussed this with @mohsen-deriv and we have to keep it like this for the time being until the icon size is fixed.